package/MDAnalysis/core/topology.py

```diff
@@ -113,6 +113,9 @@ def make_downshift_arrays(upshift, nparents):
     .. warning:: This means negative indexing should **never**
                  be used with these arrays.
     """
+    if not list(upshift):
```

Comment:
casting to a `list` might be a performance issue for large arrays. Perhaps check explicitly for `None`, `len()` and whatever else is needed. Or use a `try/except`.

Reply:
Switched to checking the length of the list/array.